components/script/script_thread.rs

```diff
@@ -693,6 +694,11 @@ pub struct ScriptThread {
 
     /// Code is running as a consequence of a user interaction
     is_user_interacting: Cell<bool>,
+
+    /// The MediaSessions known by this thread, if any.
+    /// There can only be one active MediaSession.
+    /// The constellation has the BrowsingContextId of the active MediaSession, if any.
+    media_sessions: DomRefCell<HashMap<TopLevelBrowsingContextId, Dom<MediaSession>>>,
```

Comment:
Have you considered using the `BrowsingContextId` vs the top-level for the key? The spec seem to say that each browsing-context, whether it is a top-level or a nested one, can have a session, see https://w3c.github.io/mediasession/#media-session-routing.